File: automator/src/main/java/com/stardust/automator/filter/PackageNameFilter.java
package com.stardust.automator.filter;

import com.stardust.automator.UiObject;

/**
 * Created by Stardust on 2017/3/9.
 */

public class PackageNameFilter {

    private static final KeyGetter PACKAGE_NAME_GETTER = new KeyGetter() {
        @Override
        public String getKey(UiObject nodeInfo) {
            CharSequence charSequence = nodeInfo.getPackageName();
            return charSequence == null ? null : charSequence.toString();
        }

        @Override
        public String toString() {
            return "packageName";
        }
    };

    public static ListFilter equals(String text) {
        return new StringEqualsFilter(text, PACKAGE_NAME_GETTER);
    }

    public static ListFilter contains(String str) {
        return new StringContainsFilter(str, PACKAGE_NAME_GETTER);
    }

    public static ListFilter startsWith(String prefix) {
        return new StringStartsWithFilter(prefix, PACKAGE_NAME_GETTER);
    }

    public static ListFilter endsWith(String suffix) {
        return new StringEndsWithFilter(suffix, PACKAGE_NAME_GETTER);
    }

    public static ListFilter matches(String regex) {
        return new StringMatchesFilter(regex, PACKAGE_NAME_GETTER);
    }

    private PackageNameFilter() {

    }
}

File: automator/src/main/java/com/stardust/automator/filter/PackageNameFilter.kt
package com.stardust.automator.filter

import com.stardust.automator.UiObject

/**
 * Created by Stardust on 2017/3/9.
 */

object PackageNameFilter {

    private val PACKAGE_NAME_GETTER = object : KeyGetter {
        override fun getKey(nodeInfo: UiObject): String? {
            val charSequence = nodeInfo.packageName
            return charSequence?.toString()
        }

        override fun toString(): String {
            return "packageName"
        }
    }

    fun equals(text: String): Filter {
        return StringEqualsFilter(text, PACKAGE_NAME_GETTER)
    }

    fun contains(str: String): Filter {
        return StringContainsFilter(str, PACKAGE_NAME_GETTER)
    }

    fun startsWith(prefix: String): Filter {
        return StringStartsWithFilter(prefix, PACKAGE_NAME_GETTER)
    }

    fun endsWith(suffix: String): Filter {
        return StringEndsWithFilter(suffix, PACKAGE_NAME_GETTER)
    }

    fun matches(regex: String): Filter {
        return StringMatchesFilter(regex, PACKAGE_NAME_GETTER)
    }
}
